certora/specs/SocialRecoveryModule.spec

```
@@ -436,3 +436,24 @@ rule canFinalizeRecoveryOnlyAfterDelayPeriod(env e) {
 
     assert success => require_uint64(e.block.timestamp) >= recoveryTimestamp, "Recovery finalized before delay period";
 }
+
+// Recovery can be finalized by anyone. But the success of the transaction depends on the delay period.
+rule anyoneCanFinalizeRecovery(env e) {
+    uint64 executeAfter = currentContract.recoveryRequests[safeContract].executeAfter;
+
+    requireInitiatedRecovery(safeContract);
+
+    require e.msg.value == 0;
+    require safeContract.getOwners().length > 0;
+    // The contract doesn't allow guardians as new owners, hence we require it. We cannot use `isGuardian` due to use of quantifiers.
+    require forall uint256 i.
+        !(currentContract.recoveryRequests[safeContract].newOwners[i] != SENTINEL() &&
+        currentContract.entries[safeContract].guardians[currentContract.recoveryRequests[safeContract].newOwners[i]] != 0);
+
+    currentContract.finalizeRecovery@withrevert(e, safeContract);
+
+    bool success = !lastReverted;
+
+    assert (success && require_uint64(e.block.timestamp) >= executeAfter) ||
+        (!success && require_uint64(e.block.timestamp) < executeAfter);
```

I think this is logically equivalent to:
```suggestion
    assert success <=> require_uint64(e.block.timestamp) >= executeAfter;
```

since it essentially does (a && x) || (!a && !x)